android/app/src/main/java/com/microsoft/codepush/react/CodePushTelemetryManager.java

```diff
@@ -113,13 +113,15 @@ public WritableMap getUpdateReport(WritableMap currentPackage) {
     }
 
     public void recordStatusReported(ReadableMap statusReport) {
-        if (statusReport.hasKey(STATUS_KEY) && DEPLOYMENT_SUCCEEDED_STATUS.equals(statusReport.getString(STATUS_KEY))) {
-            if (statusReport.hasKey(APP_VERSION_KEY)) {
-                saveStatusReportedForIdentifier(statusReport.getString(APP_VERSION_KEY));
-            } else if (statusReport.hasKey(PACKAGE_KEY)) {
-                String packageIdentifier = getPackageStatusReportIdentifier(statusReport.getMap(PACKAGE_KEY));
-                saveStatusReportedForIdentifier(packageIdentifier);
-            }
+        // We don't need to record rollback reports, so exit early if that's what was specified.
+        if (statusReport.hasKey(STATUS_KEY) && DEPLOYMENT_FAILED_STATUS.equals(statusReport.getString(STATUS_KEY)))
+            return;
```

Comment:
i generally avoid multi-line if statements without brackets, because
```
if (a)
    b;
```
becomes
```
if (a)
    b;
    c;
```
without much thought, which leads to issues. i would just append the `return;` on the end of the single line.